reference/test/reorder/rcm.cpp

```diff
@@ -0,0 +1,184 @@
+    std::shared_ptr<CsrMtx> id3_mtx;
+    std::shared_ptr<CsrMtx> not_id3_mtx;
+    std::unique_ptr<typename reorder_type::Factory> rcm_factory;
+    std::unique_ptr<reorder_type> reorder_op;
+};
+
+TYPED_TEST_CASE(Rcm, gko::test::ValueIndexTypes);
+
+TYPED_TEST(Rcm, CanBeCleared)
+{
+    this->reorder_op->clear();
+
+    auto reorder_op_perm = this->reorder_op->get_permutation();
+    
+    ASSERT_EQ(reorder_op_perm, nullptr);
+}
+
+TYPED_TEST(Rcm, CanBeCopied)
+{
+    using v_type = typename TestFixture::v_type;
+    auto rcm = this->rcm_factory->generate(this->id3_mtx);
+    auto rcm_copy = this->rcm_factory->generate(this->not_id3_mtx);
+
+    rcm_copy->copy_from(rcm.get());
+
+    ASSERT_EQ(rcm_copy->get_permutation()->get_const_permutation()[0], 2);
+    ASSERT_EQ(rcm_copy->get_permutation()->get_const_permutation()[1], 1);
+    ASSERT_EQ(rcm_copy->get_permutation()->get_const_permutation()[2], 0);
+}
+
+
+TYPED_TEST(Rcm, CanBeMoved)
+{
+    using v_type = typename TestFixture::v_type;
+    auto rcm = this->rcm_factory->generate(this->id3_mtx);
+    auto rcm_move = this->rcm_factory->generate(this->not_id3_mtx);
+
+    rcm->move_to(rcm_move.get());
+
+    ASSERT_EQ(rcm_move->get_permutation()->get_const_permutation()[0], 2);
+    ASSERT_EQ(rcm_move->get_permutation()->get_const_permutation()[1], 1);
+    ASSERT_EQ(rcm_move->get_permutation()->get_const_permutation()[2], 0);
+}
+
+
+TYPED_TEST(Rcm, CanBeCloned)
+{
+    using v_type = typename TestFixture::v_type;
+    auto rcm = this->rcm_factory->generate(this->id3_mtx);
+
+    auto rcm_clone = rcm->clone();
+
+    ASSERT_EQ(rcm_clone->get_permutation()->get_const_permutation()[0], 2);
+    ASSERT_EQ(rcm_clone->get_permutation()->get_const_permutation()[1], 1);
+    ASSERT_EQ(rcm_clone->get_permutation()->get_const_permutation()[2], 0);
+}
+
+TYPED_TEST(Rcm, HasSensibleDefaults)
+{
+    using v_type = typename TestFixture::v_type;
+    using reorder_type = typename TestFixture::reorder_type;
+
+    auto rcm = reorder_type::build().on(this->exec)->generate(this->id3_mtx);
+
+    ASSERT_EQ(rcm->get_parameters().construct_inverse_permutation, false);
+    ASSERT_EQ(rcm->get_parameters().strategy,
+              gko::reorder::starting_strategy::pseudo_peripheral);
+}
+
+TYPED_TEST(Rcm, CanBeCreatedWithStartingStrategy)
+{
+    using v_type = typename TestFixture::v_type;
+    using reorder_type = typename TestFixture::reorder_type;
+    auto rcm =
+        reorder_type::build()
+            .with_strategy(gko::reorder::starting_strategy::minimum_degree)
+            .on(this->exec)
+            ->generate(gko::initialize<gko::matrix::Dense<v_type>>(
+                3, {{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}},
+                this->exec));
+
+    ASSERT_EQ(rcm->get_parameters().strategy,
+              gko::reorder::starting_strategy::minimum_degree);
+}
+
+TYPED_TEST(Rcm, CanBeCreatedWithConstructInversePermutation)
+{
+    using v_type = typename TestFixture::v_type;
```

Comment:
nit: you are not using `v_type` in pretty much all tests except for `CanBeCreatedWithStartingStrategy`, so I think you should remove it where you don't use it.